client/allocrunner/networking_cni.go

```diff
@@ -134,6 +140,22 @@ func (c *cniNetworkConfigurator) Setup(ctx context.Context, alloc *structs.Alloc
 			netStatus.Address = iface.IPConfigs[0].IP.String()
 		}
 	}
+
+	if netStatus.Address == "" {
+		c.logger.Debug("no address found for sandboxed interface from CNI result, using first available")
+		var found bool
+		for _, iface := range res.Interfaces {
+			if len(iface.IPConfigs) > 0 {
+				netStatus.Address = iface.IPConfigs[0].IP.String()
```

Comment:
If a sandbox interface is found above but it lacks an address, we will continue to use its name while we use another interfaces address.

If using one interfaces name and a different interfaces address is intentional, can we comment here?